modules/afmongodb/afmongodb-worker.c

```diff
@@ -29,15 +29,62 @@
 #include "value-pairs/value-pairs.h"
 #include "scanner/list-scanner/list-scanner.h"
 
+static void
+_destroy_bulk_op_options(MongoDBDestWorker *self)
+{
+  if (self->bson_opts)
+    {
+      bson_destroy(self->bson_opts);
+      g_free(self->bson_opts);
+      self->bson_opts = NULL;
+    }
+  if (self->write_concern)
+    {
+      mongoc_write_concern_destroy(self->write_concern);
+      self->write_concern = NULL;
+    }
+}
+
+static void
+_compose_bulk_op_options(MongoDBDestWorker *self)
+{
+  MongoDBDestDriver *owner = (MongoDBDestDriver *) self->super.owner;
+
+  if (owner->mongoc_opt_use_bulk && (owner->mongoc_opt_unordered || owner->mongoc_opt_unacknowledged))
+    {
+      self->bson_opts = g_new0(bson_t, 1);
+      bson_t def_opts = BSON_INITIALIZER;
+      *self->bson_opts = def_opts;
+
+      if (owner->mongoc_opt_unordered)
+        BSON_APPEND_BOOL (self->bson_opts, "ordered", false);
+
+      if (owner->mongoc_opt_unacknowledged)
+        {
+          self->write_concern = mongoc_write_concern_new();
```

Comment:
`write_concern` is not a bulk-related option. Let's construct it separately from bulk options.

Reply:
actually it is not ONLY bulk related, but I see your point, will separate them